### PR TITLE
docs: rearrange docs and content

### PR DESCRIPTION
**Pull request by max-cx**
## What does this pull request change
Considering the fact that we have new UG sections under **User onboarding**, the following content appears out of place and needs some editing:
https://www.eclipse.org/che/docs/next/end-user-guide/using-developer-workspaces/
https://www.eclipse.org/che/docs/next/end-user-guide/first-time-contributors/
https://www.eclipse.org/che/docs/next/end-user-guide/benefits-of-pull-requests-review-in-che/

image

## What issues does this pull request fix or reference
RHDEVDOCS-3694

## Specify the version of the product this pull request applies to
7.46

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [x] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/pages/developer-workspaces.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+:navtitle: Using developer workspaces
```
Comment (themr0c) [inline modules/end-user-guide/pages/developer-workspaces.adoc]: Add a content type

```suggestion
:_content-type: CONCEPT
:navtitle: Using developer workspaces
```

File: modules/end-user-guide/pages/developer-workspaces.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+:keywords: user-guide, workspaces-overview
+:page-aliases: .:using-developer-workspaces, .:workspaces-overview, .:what-are-workspaces.html, .:creating-and-configuring-a-new-workspace, creating-and-configuring-a-new-workspace, authenticating-on-scm-server-with-a-personal-access-token, navigating-che, navigating-che-using-the-dashboard, importing-certificates-to-browsers, creating-a-workspace-from-local-devfile-using-chectl, importing-kubernetes-applications-into-a-workspace, workspaces-overview, configuring-a-workspace-with-dashboard, creating-a-workspace-from-a-code-sample, creating-a-workspace-from-a-template-devfile, creating-a-workspace-from-remote-devfile, running-a-workspace-with-dashboard, remotely-accessing-workspaces, importing-the-source-code-of-a-project-into-a-workspace
```
Comment (themr0c) [inline modules/end-user-guide/pages/developer-workspaces.adoc:4]: As you are renaming the page, add the old page name in the aliases.

```suggestion
:page-aliases: .:using-developer-workspaces, .:workspaces-overview, .:what-are-workspaces.html, .:creating-and-configuring-a-new-workspace, creating-and-configuring-a-new-workspace, authenticating-on-scm-server-with-a-personal-access-token, navigating-che, navigating-che-using-the-dashboard, importing-certificates-to-browsers, creating-a-workspace-from-local-devfile-using-chectl, importing-kubernetes-applications-into-a-workspace, workspaces-overview, configuring-a-workspace-with-dashboard, creating-a-workspace-from-a-code-sample, creating-a-workspace-from-a-template-devfile, creating-a-workspace-from-remote-devfile, running-a-workspace-with-dashboard, remotely-accessing-workspaces, importing-the-source-code-of-a-project-into-a-workspace, using-developer-workspaces
```

Fix also the xref to point to the new page, see htmltest error: 

docs/next/hosted-che/hosted-che/index.html
  hash does not exist --- docs/next/hosted-che/hosted-che/index.html --> #end-user-guide:using-developer-workspaces.adoc

**Comment by max-cx** [comment]
Closing this PR to avoid spending time on fixing the branch.
Opening https://github.com/eclipse-che/che-docs/pull/2309 afresh.

File: modules/end-user-guide/pages/developer-workspaces.adoc (hunk 2 above)
Comment (max-cx) [reply modules/end-user-guide/pages/developer-workspaces.adoc:4]: Thank you, Fabrice.
I've applied this.
Apologies for the nuisance: next time, I'll be more careful.

**Pull request closed without merging**